commands/new_function_test.go

```diff
@@ -5,37 +5,334 @@ package commands
 
 import (
 	"os"
+	"reflect"
 	"regexp"
 	"testing"
 
+	"github.com/openfaas/faas-cli/stack"
 	"github.com/openfaas/faas-cli/test"
 )
 
-func Test_new(t *testing.T) {
-	//TODO activate the teardown when PR#87 is merged defer teardown()
-	funcName := "test-function"
+const SuccessMsg = `(?m:Function created in folder)`
+const InvalidYAMLMsg = `is not valid YAML`
+const InvalidYAMLMap = `map is empty`
+const LangNotExistsOutput = `(?m:is unavailable or not supported.)`
 
-	// Symlink template directory at root to current directory to avoid re-downloading templates
-	os.Symlink("../template", "template")
+type NewFunctionTest struct {
+	title       string
+	funcName    string
+	funcLang    string
+	file        string
+	expectedMsg string
+}
+
+var NewFunctionTests = []NewFunctionTest{
+	{
+		title:       "new_1",
+		funcName:    "new-test-1",
+		funcLang:    "ruby",
+		file:        "",
+		expectedMsg: SuccessMsg,
+	},
+	{
+		title:       "new_2",
+		funcName:    "new-test-2",
+		funcLang:    "dockerfile",
+		file:        "",
+		expectedMsg: SuccessMsg,
+	},
+}
+
+var AppendFunctionTests = []NewFunctionTest{
+	{
+		title:       "new_append_1",
+		funcName:    "new-test-append-1",
+		funcLang:    "python",
+		file:        "append1.yml",
+		expectedMsg: SuccessMsg,
```

Comment:
In this example it may be OK to have the const / remove the repetition.

Reply:
You mean only have one test case for `AppendFunctionTests`? Or something like, one positive one negative test case?